library/std/src/sys/backtrace.rs

```diff
@@ -12,13 +12,15 @@ use crate::sync::{Mutex, PoisonError};
 /// Max number of frames to print.
 const MAX_NB_FRAMES: usize = 100;
 
-pub fn lock() -> impl Drop {
+pub(crate) fn lock() -> impl Drop {
     static LOCK: Mutex<()> = Mutex::new(());
     LOCK.lock().unwrap_or_else(PoisonError::into_inner)
 }
 
 /// Prints the current backtrace.
-pub fn print(w: &mut dyn Write, format: PrintFmt) -> io::Result<()> {
+///
+/// SAFETY: this function is not Sync. The caller must hold a mutex lock, or there must be only one thread in the program.
+pub(crate) unsafe fn print(w: &mut dyn Write, format: PrintFmt) -> io::Result<()> {
```

Comment:
so uh silly question maybe, I guess it's a few more lines of code but why isn't `backtrace::print` a function implemented on the lock that `backtrace::lock` returns, given that we should be holding that specific lock...?

Reply:
no particular reason. i can implement that tomorrow.

Reply:
and by tomorrow I assume you mean Monday, but of course. :sparkling_heart:

Reply:
done!